nginx.tmpl

```diff
@@ -668,7 +668,7 @@ server {
         {{ template "upstream" (dict "globals" $globals "Path" $path "VPath" $vpath) }}
     {{- end }}
 
-    {{- if and $vhost.cert_ok (eq $vhost.https_method "redirect") }}
+    {{- if and (or $vhost.cert_ok $globals.default_cert_ok) (or (eq $vhost.https_method "redirect") (eq $vhost.https_method "nohttp")) }}
```

Comment:
I'm not certain about silently running an HTTP server for the sole purpose of the acme-challenge when using `nohttp`.

To me using `nohttp` when wanting to obtain ACME TLS certificate with the HTTP challenge should fail without exception, [because not blocking port 80 at all is part of LE's best practice](https://letsencrypt.org/docs/allow-port-80/) (with very sound arguments imho), and the[ RFC8555 section 8.3](https://www.rfc-editor.org/rfc/rfc8555.html#section-8.3) itself indicate that first connection of an ACME HTTP challenge must happen on port 80, which make it incompatible with attempt to block this port by design:

> 3. Dereference the URL using an HTTP GET request.  This request MUST be sent to TCP port 80 on the HTTP server

I think we should instead update the documentation to warn people that `nohttp` will be incompatible with the ACME HTTP challenge, [like in acme-companion's documentation](https://github.com/nginx-proxy/acme-companion?tab=readme-ov-file#requirements).

Explicit use of `nohttp` by users should be done knowingly and observed entirely by nginx-proxy.

Reply:
The use case of people absolutely having to block port 80 yet still wanting ACME TLS certificate should rather be handled by making acme-companion compatible with DNS challenge (the next "big" feature I'd like to get merged).

Reply:
There is a very simple solution when rigorously no http server is wanted: do not bind port 80.

What makes me uncomfortable is acme-companion fiddling into the nginx configuration. It shouldn't have to.

How about enabling acme challenge if and only if:
* the nginx-proxy container has `ACME_CHALLENGE=true`
* or the app container has `ACME_CHALLENGE=true`?
* or the app container defines the `LETSENCRYPT_HOST` variable,

If you agree on none of these proposals, I'd be OK with not enabling acme challenge when `HTTPS_METHOD=nohttp`.

What do you think?

Reply:
> But I might be missing valid use cases of using `nohttp` and still wanting to be able to complete ACME HTTP challenge, do you have some examples ?

Unfortunately I don't know any use case for `HTTPS_METHOD=nohttp`.

> If we decide to enable ACME challenge when `HTTPS_METHOD=nohttp`, I'd go with the third solution (based on wether the app container defines the `LETSENCRYPT_HOST` variable or not).

If you want to give it more thinking I could split this PR in two:
1. Handle acme challenge but for `HTTPS_METHOD=nohttp`
2. When `HTTPS_METHOD=no_http` handle acme challenge only if the app container defines `LETSENCRYPT_HOST`.

Actually I'm not sure (2) is needed, So I'll go for the split.

Reply:
That was maybe unclear but the only gripe I had was with `nohttp` (so only with 2.), the rest of the PR make sense to me and I was about to ping you on this issue. I don't have any issue merging a PR for 1.  👍

Reply:
> Edit : I think we might have to enable the challenge only for app containers that have the `LETSENCRYPT_HOST` variable set,

This might break existing use cases where acme-companion is not used but acme challenge is needed. We could do the other way around: do not enable acme challenge when `NO_ACME_CHALLENGE=true`.

Reply:
> EDIT: But I'd prefer to implement this via another PR.

This is #2447.